src/storage/src/source/generator.rs

```diff
@@ -106,8 +106,14 @@ impl SourceConnectionBuilder for LoadGeneratorSourceConnection {
             .by_seed(mz_ore::now::SYSTEM_TIME.clone(), None);
 
         // Skip forward to the requested offset.
+        let mut transactions = rows
+            .by_ref()
+            .filter(|(_, typ, _, _)| matches!(typ, GeneratorMessageType::Finalized));
         for _ in 0..offset.offset {
-            rows.next();
+            assert!(
```

Comment:
Is it safe to depend on rust not optimizing this away in some build configurations?

Reply:
This is a great point that I didn't think about! It's probably safe but I'm going to pull the side effects out of the assert to be certain

Reply:
Fixed!

Reply:
(FWIW, `assert!` is [guaranteed to be run in every possible build config](https://doc.rust-lang.org/std/macro.assert.html#uses). Doesn't hurt to move this out, though!)

Reply:
That's good to know! thanks for the link